### PR TITLE
Check old password during password change and add missing CLI commands

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes
As an extra security measure, this PR always requires that the current password value be supplied during a password change.

Additional related changes:
* implement a separate `zenml user change-password` CLI command for changing the password for the current user
* warn about passing passwords using command-line arguments in `zenml user change-password`, `zenml user create` and `zenml user update`
* warn about using username/password to connect to a ZenML server and suggest using the web login flow or a service account API key instead
* allow a user account to be activated and deactivated through `zenml user update`
* implement the `zenml user deactivate` CLI command to be used to reset other user accounts through the CLI (by admins only).

**IMPORTANT**: this change invalidates the current ZenML Dashboard password change support. A dashboard update is required to pass the current password to the call made to the API (handled with https://github.com/zenml-io/zenml-dashboard/pull/556)

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [x] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/cli/user_management.py [hunk 1]
```diff
@@ -276,12 +287,130 @@ def update_user(
...
+            "Please consider using the prompt option."
+        )
+
+    if old_password is None:
+        old_password = click.prompt(
+            f"Current password for user {active_user.name}",
+            hide_input=True,
+        )
+    if password is None:
+        password = click.prompt(
```
Comment (bcdurak) [inline src/zenml/cli/user_management.py:342]: In many cases where I had to reset my password, I had to enter the new password twice. It is a nice mechanism to fight against the typos, perhaps we can also add it here?

**Review by bcdurak — APPROVED** [review]
Thank you! The changes look good. 

I just added a small comment for a minor QoL improvement (feel free to ignore).

I had the same question as @avishniakov. I am guessing the changes will go in once the dashboard is ready.

**Comment by stefannica** [comment]
> Thanks, nice feature! Do we have dashboard changes ready? If not I would pause merging this, so it is not in the upcoming release, until Dashboard has this. @Cahllagerfeld

@avishniakov yes, the dashboard changes have already been merged: https://github.com/zenml-io/zenml-dashboard/pull/556

File: src/zenml/cli/user_management.py (hunk 1 above)
Comment (stefannica) [reply src/zenml/cli/user_management.py:342]: Done

**Pull request merged**